File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/channels/ChannelsArchiveRequest.java
package com.github.seratch.jslack.api.methods.request.channels;

import com.github.seratch.jslack.api.methods.SlackApiRequest;

public class ChannelsArchiveRequest implements SlackApiRequest {

    /**
     * Authentication token. Requires scope: `channels:write`
     */
    private String token;

    /**
     * Channel to archive
     */
    private String channel;

    ChannelsArchiveRequest(String token, String channel) {
        this.token = token;
        this.channel = channel;
    }

    public static ChannelsArchiveRequestBuilder builder() {
        return new ChannelsArchiveRequestBuilder();
    }

    public String getToken() {
        return this.token;
    }

    public String getChannel() {
        return this.channel;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public void setChannel(String channel) {
        this.channel = channel;
    }

    public static class ChannelsArchiveRequestBuilder {
        private String token;
        private String channel;

        ChannelsArchiveRequestBuilder() {
        }

        public ChannelsArchiveRequest.ChannelsArchiveRequestBuilder token(String token) {
            this.token = token;
            return this;
        }

        public ChannelsArchiveRequest.ChannelsArchiveRequestBuilder channel(String channel) {
            this.channel = channel;
            return this;
        }

        public ChannelsArchiveRequest build() {
            return new ChannelsArchiveRequest(token, channel);
        }

        public String toString() {
            return "ChannelsArchiveRequest.ChannelsArchiveRequestBuilder(token=" + this.token + ", channel=" + this.channel + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/channels/ChannelsArchiveRequest.kt
package work.socialhub.kslack.api.methods.request.channels

import work.socialhub.kslack.api.methods.FormRequest
import work.socialhub.kslack.api.methods.SlackApiRequest
import kotlin.js.JsExport

@JsExport
class ChannelsArchiveRequest(
    /** Authentication token. Requires scope: `channels:write` */
    override var token: String?,
    /** Channel to archive */
    var channel: String?
) : SlackApiRequest, FormRequest {

    override fun toMap(): Map<String, Any> {
        return mutableMapOf<String, Any>().also {
            it.addParam("channel", channel)
        }
    }
}